src/learn/upgrade.md

```diff
@@ -0,0 +1,118 @@
+# Upgrade procedure
+
+To be able to do an upgrade, first there needs to be an approved software upgrade proposal. Also there are two types of upgrade procedures. The first is being used when there are no breaking changes and the second - when there is.
+
+# Preparation
+
+The preparation steps are required in both cases - with or without breaking changes.
+
+## Software upgrade proposal
+
+A software upgrade proposal is first being submitted. After being submitted it enters into a deposit period, during which a certain amount of tokens need to be deposited into it, so the voting period can start. After the voting period starts, only validators can vote and approve it. If the proposal is passed, the network stops at a specified block height, until the upgrade is made and network is restarted.
+
+### Submitting a software upgrade proposal
+
+A proposal can be submitted with the following command in a node terminal:
+
+```bash
+cudos-noded tx gov submit-proposal software-upgrade <proposal_name> --upgrade-height <block_at_which_to_stop> --from <wallet_name> --deposit <amount_to_deposit_with_denomination> --title <proposal_title> --description <proposal_description> --keyring-backend <os or file or test> --chain-id <chain_id> -y
+```
+
+### Depositing to a proposal
+
+To get the ID of the proposal that we want to deposit into, enter the following command:
+
+```bash
+cudos-noded q gov proposals
+```
+
+This will return a list with all the proposals in which we can find the one we need and find the field "**proposal_id**" for it. Then to deposit funds into it, enter the following command:
+
+```bash
+cudos-noded tx gov deposit <proposal_id> <amount_with_denomination> --from <wallet_name> --keyring-backend <os or file or test> --chain-id <chain_id> -y
+```
+
+If the funds are enough, the proposal should enter in a voting status, which can again be seen with the command we used for the proposal ID.
+
+### Voting a proposal
+
+To vote with yes to a proposal, use the following command:
+
+```bash
+cudos-noded tx gov vote <proposal_id> yes --from <walled_name> --keyring-backend <os or file or test> --chain-id <chain_id> -y
+```
+
+If enough votes with "yes" are sent, the proposal will be approved and the network will stop at the specified block height or time.
+
+The chain will stop at specified height. When the chain stops we need to stop the nodes as well.
+
+# Soft upgrade (without breaking changes)
+
+It is being done by "in-place migration", described [here](https://docs.cosmos.network/master/core/upgrade.html).
+
+In short - after the network stops we have to pull and build the new binary and start the chain again.
+
+# Hard fork (with breaking changes)
+
+After the network stops we must:
```

Comment:
Who's the 'we' in this context? Does each individual Validator have to run this simultaneously?

Reply:
All validators need to go through the process. I've made some updates that hopefully will make help clear the confusion.